commit 925a3225708c40c8cd58a7475c5a6d3bd89db01d
Author: R David Murray <rdmurray@bitdance.com>
Date:   Sat Jun 29 18:40:53 2013 -0400

    #18155: Regex-escape delimiter, in case it is a regex special char.
    
    Patch by Vajrasky Kok, with slight modification to the tests by me.

diff --git a/csv_module.py b/csv_module.py
index 8dfc77e310..da3bc44e7a 100644
--- a/csv_module.py
+++ b/csv_module.py
@@ -264,8 +264,9 @@ class Sniffer:
 
         # if we see an extra quote between delimiters, we've got a
         # double quoted format
-        dq_regexp = re.compile(r"((%(delim)s)|^)\W*%(quote)s[^%(delim)s\n]*%(quote)s[^%(delim)s\n]*%(quote)s\W*((%(delim)s)|$)" % \
-                               {'delim':delim, 'quote':quotechar}, re.MULTILINE)
+        dq_regexp = re.compile(
+                               r"((%(delim)s)|^)\W*%(quote)s[^%(delim)s\n]*%(quote)s[^%(delim)s\n]*%(quote)s\W*((%(delim)s)|$)" % \
+                               {'delim':re.escape(delim), 'quote':quotechar}, re.MULTILINE)
 
 
 
